acceptance/tests/restart_host_run_puppet.rb

```diff
@@ -48,7 +48,14 @@
 
   step "restart each agent" do
     applicable_agents.each do |agent|
-      agent.reboot
+      # Work-around reboot hanging on reply (BKR-971)
+      begin
+        Timeout.timeout(5) do
+          agent.reboot
+        end
```

Comment:
Ah, in that case, I'd probably execute the `reboot` call in a separate thread.

Also it looks like beaker tried to workaround the issue in https://github.com/puppetlabs/beaker/commit/f96a15aa3d0f701618bf1e63f78a40f2329c4298#diff-3556c18d916e298ae0708391158c86d0R6, but I don't think adding the `sleep(30)` is going to help anything, since beaker is blocked waiting for the response to the command.